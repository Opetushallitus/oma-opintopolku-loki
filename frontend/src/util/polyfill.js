import 'core-js/features/promise'
import 'core-js/features/string/starts-with'
