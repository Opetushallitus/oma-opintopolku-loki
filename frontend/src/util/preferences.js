import Cookies from 'js-cookie'

export const lang = Cookies.get('lang') || 'fi'
