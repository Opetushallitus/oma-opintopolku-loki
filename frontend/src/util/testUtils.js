/* istanbul ignore file */
import renderer from 'react-test-renderer'

export function matchesSnapshot (component) {
  const renderedComponent = renderer.create(component)
  const componentJSON = renderedComponent.toJSON()
  expect(componentJSON).toMatchSnapshot()
  renderedComponent.unmount()
}
