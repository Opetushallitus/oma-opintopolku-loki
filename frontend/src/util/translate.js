import Cookies from 'js-cookie'

export default key => {
  const lang = Cookies.get('lang') || 'fi'
  const translations = window.translationsMap || {}
  const versions = translations[key] || {}
  const translated = versions[lang]

  if (!translated) {
    console.error(`Translation missing for language ${lang}: ${key}`)
    return key
  }

  return translated
}
