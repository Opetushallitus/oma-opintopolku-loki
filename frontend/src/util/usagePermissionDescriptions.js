import t from 'util/translate'

import permissionsMap from 'Resources/mapping/usagePermissionDescriptions'

export const getTranslatedUsagePermissionDescription = organizationOid => {
  const permission = permissionsMap[organizationOid] || permissionsMap.default
  return t(permission)
}

export const isMydataPartner = organizationOid => {
  return Boolean(permissionsMap[organizationOid])
}
