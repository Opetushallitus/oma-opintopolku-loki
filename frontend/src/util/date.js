import moment from 'moment'
import { lang } from 'util/preferences'

moment.locale(lang)

export const isoStringToDate = isoString => moment(isoString).format('l, LT')
