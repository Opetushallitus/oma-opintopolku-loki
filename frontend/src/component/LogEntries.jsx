import React from 'react'
import { lensProp, omit, over, uniq } from 'ramda'
import Query from 'http/Query'
import LogEntry from 'component/LogEntry'
import { lang } from 'util/preferences'
import { AlertText } from '../ui/typography'
import t from 'util/translate'

const getTranslatedName = organizationName => organizationName[lang] || ''
const nameLens = lensProp('organizationName')

const LogEntries = () => (
  <Query url='logs'>
    {({ data, error, pending }) => {
      if (error) return <AlertText>{t`Tietojen hakemisessa tapahtui virhe.`}</AlertText>
      if (pending) return <div>{t`Tietoja haetaan`}</div>

      const organizations = uniq(data
        .map(omit(['timestamp']))
        .map(over(nameLens, getTranslatedName)))

      return organizations.map(({ organizationOid, organizationName }) =>
        <LogEntry
          key={organizationOid}
          organizationOid={organizationOid}
          organizationName={organizationName}
        />
      )
    }}
  </Query>
)

export default LogEntries
