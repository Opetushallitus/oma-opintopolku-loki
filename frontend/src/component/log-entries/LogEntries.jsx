import React from 'react'
import PropTypes from 'prop-types'
import { compose, withHandlers, withState } from 'recompose'
import styled from 'styled-components'
import { descend, identity, sort } from 'ramda'
import t from 'util/translate'
import { isoStringToDate } from 'util/date'
import constants from 'ui/constants'
import { Bold } from 'ui/typography'
import ButtonSmall from 'component/generic/widget/ButtonSmall'
import DateList from 'component/generic/widget/DateList'

const NUM_INITIAL_DATES = 10
const NUM_SHOW_MORE_DATES = 10
const MARGIN = '1rem'

const Container = styled.div`
  margin: ${MARGIN} -${MARGIN} 0 -${MARGIN};
  padding: ${MARGIN};
  background-color: ${({ showList }) => showList ? constants.color.background.neutralLight : 'none'};
`

const TopRow = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: row-reverse;
  margin-left: ${MARGIN};
  margin-bottom: 0.5rem;
`

const BottomRow = styled.div`
  margin-left: ${MARGIN};
`

const addListBehaviors = compose(
  withState('showDates', 'setShowDates', false),
  withState('numDatesShown', 'setNumDatesShown', NUM_INITIAL_DATES),
  withHandlers({
    toggleShowDates: ({ setShowDates }) => () => setShowDates(show => !show),
    showMoreDates: ({ setNumDatesShown }) => () => setNumDatesShown(n => n + NUM_SHOW_MORE_DATES)
  })
)

const LogEntries = addListBehaviors(({ timestamps, showDates, toggleShowDates, numDatesShown, showMoreDates }) => {
  const hasMoreEntries = numDatesShown < timestamps.length
  const sortedTimestamps = sort(descend(identity), timestamps)
  const dates = sortedTimestamps.map(isoStringToDate)

  return (
    <Container showList={showDates}>
      <TopRow>
        <ButtonSmall onClick={toggleShowDates} attributes={{ 'aria-pressed': showDates }}>
          {showDates ? t`Piilota käyttökerrat` : t`Taulukko tietojen käyttökerroista`}
        </ButtonSmall>
        {showDates && (
          <Bold>
            {t`Näytetään käyttökerrat`}
            {` 1–${Math.min(numDatesShown, dates.length)} / ${dates.length}`}
          </Bold>
        )}
      </TopRow>

      {showDates && (
        <React.Fragment>
          <DateList dates={dates} numShown={numDatesShown} />

          {hasMoreEntries && (
            <BottomRow>
              <ButtonSmall onClick={showMoreDates}>{t`Näytä lisää käyttökertoja`}</ButtonSmall>
            </BottomRow>
          )}
        </React.Fragment>
      )}
    </Container>
  )
})

LogEntries.propTypes = {
  timestamps: PropTypes.array.isRequired
}

export default LogEntries
