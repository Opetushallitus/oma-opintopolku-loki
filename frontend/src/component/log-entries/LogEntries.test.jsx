import React from 'react'
import { mount } from 'enzyme'
import LogEntries from './LogEntries'
import { matchesSnapshot } from 'util/testUtils'

describe('LogEntries', () => {
  const timestamps = ['2018-09-19T12:05:26.432+03', '2018-09-20T16:01:44.543+03']

  // eslint-disable-next-line jest/expect-expect
  it('should render with empty timestamp list', () => {
    matchesSnapshot(<LogEntries timestamps={[]} />)
  })

  it('should render with non-empty timestamp list, and toggle expansion of timestamps when link is clicked', () => {
    const wrapper = mount(<LogEntries timestamps={timestamps} />)
    expect(wrapper.render()).toMatchSnapshot()

    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.render()).toMatchSnapshot()

    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.render()).toMatchSnapshot()
  })

  it('should render with timestamp list of over 10 entries, expanding 10 timestamps at a time', () => {
    const ts = Array(15).fill('2018-09-19T12:05:26.432+03')
    const wrapper = mount(<LogEntries timestamps={ts} />)

    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.render()).toMatchSnapshot()

    wrapper.find('button').at(1).simulate('click')
    expect(wrapper.render()).toMatchSnapshot()
  })
})
