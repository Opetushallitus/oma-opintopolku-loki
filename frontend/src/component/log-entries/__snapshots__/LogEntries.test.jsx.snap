// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogEntries should render with empty timestamp list 1`] = `
<div
  className="LogEntries__Container-sc-ufvf4x-0 jIcutE"
>
  <div
    className="LogEntries__TopRow-sc-ufvf4x-1 ekwEvt"
  >
    <button
      aria-pressed={false}
      className="ButtonSmall__TextButton-sc-r8zxw8-0 fKtihz"
      onClick={[Function]}
    >
      Näytä tietojen käyttökerrat
    </button>
  </div>
</div>
`;

exports[`LogEntries should render with non-empty timestamp list, and toggle expansion of timestamps when link is clicked 1`] = `
<div
  class="LogEntries__Container-sc-ufvf4x-0 jIcutE"
>
  <div
    class="LogEntries__TopRow-sc-ufvf4x-1 ekwEvt"
  >
    <button
      aria-pressed="false"
      class="ButtonSmall__TextButton-sc-r8zxw8-0 fKtihz"
    >
      Näytä tietojen käyttökerrat
    </button>
  </div>
</div>
`;

exports[`LogEntries should render with non-empty timestamp list, and toggle expansion of timestamps when link is clicked 2`] = `
<div
  class="LogEntries__Container-sc-ufvf4x-0 eRuOKW"
>
  <div
    class="LogEntries__TopRow-sc-ufvf4x-1 ekwEvt"
  >
    <button
      aria-pressed="true"
      class="ButtonSmall__TextButton-sc-r8zxw8-0 fKtihz"
    >
      Piilota käyttökerrat
    </button>
    <b
      class="typography__Bold-sc-1yq3b4j-6 haiWbX"
    >
      Näytetään käyttökerrat 1–2 / 2
    </b>
  </div>
  <table
    class="DateList__DateTable-sc-9hn0em-0 dOMHeP"
  >
    <tbody>
      <tr>
        <td>
          20.9.2018 16.01
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`LogEntries should render with non-empty timestamp list, and toggle expansion of timestamps when link is clicked 3`] = `
<div
  class="LogEntries__Container-sc-ufvf4x-0 jIcutE"
>
  <div
    class="LogEntries__TopRow-sc-ufvf4x-1 ekwEvt"
  >
    <button
      aria-pressed="false"
      class="ButtonSmall__TextButton-sc-r8zxw8-0 fKtihz"
    >
      Näytä tietojen käyttökerrat
    </button>
  </div>
</div>
`;

exports[`LogEntries should render with timestamp list of over 10 entries, expanding 10 timestamps at a time 1`] = `
<div
  class="LogEntries__Container-sc-ufvf4x-0 eRuOKW"
>
  <div
    class="LogEntries__TopRow-sc-ufvf4x-1 ekwEvt"
  >
    <button
      aria-pressed="true"
      class="ButtonSmall__TextButton-sc-r8zxw8-0 fKtihz"
    >
      Piilota käyttökerrat
    </button>
    <b
      class="typography__Bold-sc-1yq3b4j-6 haiWbX"
    >
      Näytetään käyttökerrat 1–10 / 15
    </b>
  </div>
  <table
    class="DateList__DateTable-sc-9hn0em-0 dOMHeP"
  >
    <tbody>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="LogEntries__BottomRow-sc-ufvf4x-2 ksUUKr"
  >
    <button
      class="ButtonSmall__TextButton-sc-r8zxw8-0 fKtihz"
    >
      Näytä lisää käyttökertoja
    </button>
  </div>
</div>
`;

exports[`LogEntries should render with timestamp list of over 10 entries, expanding 10 timestamps at a time 2`] = `
<div
  class="LogEntries__Container-sc-ufvf4x-0 eRuOKW"
>
  <div
    class="LogEntries__TopRow-sc-ufvf4x-1 ekwEvt"
  >
    <button
      aria-pressed="true"
      class="ButtonSmall__TextButton-sc-r8zxw8-0 fKtihz"
    >
      Piilota käyttökerrat
    </button>
    <b
      class="typography__Bold-sc-1yq3b4j-6 haiWbX"
    >
      Näytetään käyttökerrat 1–15 / 15
    </b>
  </div>
  <table
    class="DateList__DateTable-sc-9hn0em-0 dOMHeP"
  >
    <tbody>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
      <tr>
        <td>
          19.9.2018 12.05
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
